tests/test_plugin_loader_edge_cases.py

```
@@ -265,26 +266,47 @@ def test_load_plugins_from_directory_sys_path_manipulation_error(self):
+            def side_effect(path, **_kwargs):
+                """
+                Simulate a filesystem access check that denies access for restricted paths.
+
+                Parameters:
+                        path (str): Filesystem path to check.
+
+                Raises:
+                        PermissionError: If `path` starts with "/restricted".
+                """
+                if path.startswith("/restricted"):
+                    raise PermissionError("Permission denied")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)